### PR TITLE
Fix multiple lint errors (shellcheck) and make some refactoring

### PR DESCRIPTION
**Pull request by vutny**
Plus style and logic corrections, mostly for FreeBSD and OpenBSD.

### CODE REVIEW

File: bootstrap-salt.sh [hunk 1]
```diff
@@ -1587,6 +1587,57 @@ movefile() {
 
 
 #---  FUNCTION  -------------------------------------------------------------------------------------------------------
+#          NAME:  linkfile
+#   DESCRIPTION:  Simple function to create symlinks. Overrides if asked. Accepts globs.
+#----------------------------------------------------------------------------------------------------------------------
+linkfile() {
```
Comment (s0undt3ch) [inline bootstrap-salt.sh]: Please prefix this with `__`. The only intent is to "tell" someone reading that it's an internal function, there's no other gain from that.

File: bootstrap-salt.sh [hunk 2]
```diff
@@ -4481,18 +4544,18 @@ install_openbsd_post() {
...
+                    echo salt_${fname}_paths=\"/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin\" >>/etc/rc.conf.local
+                fi
             fi
         fi
     done
 }
 
 install_openbsd_check_services() {
-  salt --version
+    #salt --version # DEBUG?
```
Comment (s0undt3ch) [inline bootstrap-salt.sh]: Hmm, I've been way for quite a while and I haven't seen that go in... Should be debug info...

**Comment by s0undt3ch** [comment]
The `cd .... || exit` lint complaints can probably be disabled. 

Thanks for cleaning up the script.

You can see the lint information right? https://jenkins-testing.saltstack.com/job/bootstrap/job/pr/job/768/job/lint/1/violations/file/bootstrap-salt.sh/

**Comment by s0undt3ch** [comment]
Dam, this Jenkins setup is not right(it's setting commit status here form another PR)....

**Comment by vutny** [comment]
Thanks @s0undt3ch 
I saw that Jenkins status, looked suspicious for me too :-) I'll fix function name and other warnings.

**Comment by vutny** [comment]
@s0undt3ch I've fixed all lint warnings from previous build, but the last time it showed up something totally unrelated to my version (obviously wrong PR 666): https://jenkins-testing.saltstack.com/job/bootstrap/job/pr/job/666/job/lint/34/violations/file/bootstrap-salt.sh/

I guess this PR is OK for now. Could you please take a look?

**Comment by vutny** [comment]
@s0undt3ch Do you have some time to review this PR? Thanks!

**Comment by vutny** [comment]
Ping @s0undt3ch

**Comment by s0undt3ch** [comment]
Sorry about the slowness.

**Pull request merged**